src/client/pfs_new_storage.go

```diff
@@ -89,7 +90,33 @@ func (c APIClient) GetTarConditional(repoName string, commitID string, path stri
 		}
 	}
 	return nil
+}
 
+// ListFileNS returns info about all files in a Commit under path.
+func (c APIClient) ListFileNS(repoName string, commitID string, path string) (finfos []*pfs.FileInfoNewStorage, retErr error) {
```

Comment:
This should be callback based similar to `GetTarConditional`. This will use too much memory for a large number of files (plus you have to wait for the full set of file infos before returning any). The callback pattern allows us to take advantage of the fact that list file rpc streams the file infos.